src/main/java/minicraft/entity/Arrow.java

```diff
@@ -67,10 +68,17 @@ public void tick() {
 		List<Entity> entitylist = level.getEntitiesInRect(new Rectangle(x, y, 0, 0, Rectangle.CENTER_DIMS));
 		boolean criticalHit = random.nextInt(11) < 9;
 		for (Entity hit : entitylist) {
+			if(hit instanceof ObsidianKnight)this.remove(); //just remove it
 			if (hit instanceof Mob && hit != owner) {
 				Mob mob = (Mob) hit;
 				int extradamage = (hit instanceof Player ? 0 : 3) + (criticalHit ? 0 : 1);
-				mob.hurt(owner, damage + extradamage, dir);
+				if(hit instanceof ObsidianKnight){
+
+					if(mob.health<2500) //if we are on phase 2
```

Comment:
I'd find a way to put this logic in the ObsidianKnight class instead of the arrow class. I don't see why the arrow needs to know which phase of a monster it is hitting.

Reply:
It’s because the armored obk is immune to arrows until it’s armor breaks in phase 2.

Reply:
That was my point. To me that is logic that belongs to obk not the arrow, which is why it makes more sense to put it in that class. Another alternative is to have a property on mobs that mark them as immune to arrows/projectiles and check that in the arrow class.

Reply:
You like tnt blacklist but for arrow?

Reply:
The issue I see with blacklisting obk is that it’s immune for only phase 1 not phase 2. Which is complicated. The logic by using the obk class is a bit more complicated than using the arrow class since the arrow class has a touch trigger, makes it so much easier.

Reply:
In fact, that should be handled by Obsidian Knight instead of Arrow as the job of arrow is only attacking (action), and for obsidian knight is only receiving damages (calculation). Damages should only be handled by damage receiving.